common/txmgr/types/client.go

```diff
@@ -3,38 +3,40 @@ package types
 import (
 	"context"
 	"math/big"
+
+	"github.com/smartcontractkit/chainlink/v2/common/types"
 )
 
 // A generic client interface for communication with the RPC node
 // Every native chain must implement independently
-type Client[CHAINID any, ADDR comparable, BLOCK any, TX any, HASH comparable, TXRECEIPT any, EVENT any, EVENTOPS any] interface {
+type Client[CHAINID ID, SEQ Sequence, ADDR types.Hashable[ADDR], BLOCK any, TX any, HASH types.Hashable[HASH], TXRECEIPT any, EVENT any, EVENTOPS any] interface {
```

Comment:
Reformat this by moving each generic type into its own line, and add comment explaining each type.

type Client[
// CHAIN_ID represents the chain's id type
CHAIN_ID ID,

// SEQ represents chain's sequence type. For example, ethereum uses a nonce.
SEQ Sequence,

.
.
.
] interface {